src/System.Buffers.Primitives/System/Buffers/ReadOnlyBufferReader.cs

```diff
@@ -42,7 +44,9 @@ public BufferReader(TSequence buffer)
 
         public Position Position => _currentPosition + _index;
 
-        public ReadOnlySpan<byte> Span => _currentSpan;
+        public ReadOnlySpan<byte> CurrentSegment => _currentSpan;
+
+        public ReadOnlySpan<byte> UnreadSegment => _currentSpan.Slice(_index);
```

Comment:
This is a useful property 😄 . I think we should expose this as the RemainingSpan and we should also have RemainingBytes

Reply:
Remaining sounds like you are talking about entire buffer, especially when RemainingBytes is absolute.

Reply:
I think in general calling these two concepts "Span" has the same issue: it's not clear what the span contains. Segment is very clear: the type is a list of segments and it's one of these.